apps/publikator/components/editor/utils/keyHandlers.js

```diff
@@ -158,9 +159,12 @@ export const createSoftBreakKeyHandler =
     if (event.key !== 'Enter') return
     if (event.shiftKey === false) return
 
-    const { startBlock } = value
+    const { startBlock, startText, endOffset } = value
+
+    const isConsecutiveSoftBreak = startText?.text[endOffset - 1] === '\n' || startText?.text[endOffset] === '\n'
```

Comment:
I wonder if instead of using the offset directly, we could just check `startText.endsWith('\n')` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith ? Or is the `endOffset` not always the end of the string? Anyway, not that important.

Reply:
ah, excellent question i didn't think of that... i'll find a solution for that, too

Reply:
thanks for the `startText` suggestion, it's a good question too and I don't know the answer, so I'll try that as well.

Reply:
I checked the docs: `startText` is the whole text node where the `startCursor` is placed, while `endText` is the text node of the `endCursor`. `startOffset` and `endOffset` play a similar role for text offsets. If if the selection is collapsed (no text is selected and the cursor is just blinking) start and end cursors are the same. I changed the code so that things also work in the case the selection is active.